### PR TITLE
TW-190: notifications component

### PR DESCRIPTION
**Pull request by letier3110**

### CODE REVIEW

File: src/screens/notifications/news/news-page.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+          <View style={styles.detailsContainer}>
+            <Text style={styles.createdAt}>{formatDate(createdAt)}</Text>
+            <Text style={styles.createdAt}>•</Text>
+            <TextLink url={readInOriginalUrl}>Read In Original</TextLink>
+          </View>
+        </View>
+      </ScrollView>
+      <View style={styles.submitContainer}>
+        <ButtonsContainer>
+          <ButtonLargePrimary title="Got it" onPress={handleSubmitPress} />
```
Comment (LendiHop) [inline src/screens/notifications/news/news-page.tsx:69]: just goBack here

File: src/screens/notifications/notifications.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  const isAllSeen = useIsEveryNewsSeenSelector();
+  const news = useNewsSelector();
+  const dispatch = useDispatch();
+
+  const isShowPlaceholder = useMemo(() => news.length === 0, [news]);
+
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      dispatch(viewNewsAction(news.filter(x => x.status === StatusType.New).map(x => x.id)));
+      // dispatch(viewNewsAction(news.map(x => x.id)));
```
Comment (LendiHop) [inline src/screens/notifications/notifications.tsx:27]: rm

**Review by LendiHop — APPROVED** [review]

File: src/components/icon/icon-name.enum.ts [hunk 3]
```diff
@@ -11,6 +11,12 @@ export enum IconNameEnum {
   CycleStatusFuture = 'CycleStatusFuture',
   BookOpen = 'BookOpen',
   Terminal = 'Terminal',
+  NotificationsUnread = 'NotificationsUnread',
+  NotificationsRead = 'NotificationsRead',
+  NewsUnread = 'NewsUnread',
+  NewsRead = 'NewsRead',
+  UpdateUnread = 'UpdateUnread',
+  UpdateRead = 'UpdateRead',
```
Comment (unixvb) [inline src/components/icon/icon-name.enum.ts:19]: rework displaying orange dot separately from svg icons

File: src/interfaces/news.interface.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  Extension = 'Extension'
+}
+
+export enum StatusType {
+  New = 'New',
+  Read = 'Read',
+  Viewed = 'Viewed'
+}
+
+export interface NewsNotificationInterface {
```
Comment (unixvb) [inline src/interfaces/news.interface.ts:18]: file & interface missmatch

File: src/screens/notifications/news/news-item.tsx [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    if (type === NewsType.ApplicationUpdate) {
+      return IconNameEnum.UpdateRead;
+    }
+
+    return IconNameEnum.NewsRead;
+  }, [type]);
+
+  const handlePress = useCallback(() => {
+    navigate(ScreensEnum.NewsScreen, { id });
+  }, []);
```
Comment (unixvb) [inline src/screens/notifications/news/news-item.tsx:43]: no need in useCallback here

File: src/screens/notifications/news/news-page.tsx [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+  usePageAnalytic(ScreensEnum.Notifications, id);
+
+  return (
+    <>
+      <ScrollView>
+        <View
+          style={[
+            styles.imageWrapper,
+            {
+              width: windowWidth
```
Comment (unixvb) [inline src/screens/notifications/news/news-page.tsx:47]: could you do the same using width:100%?

File: src/screens/notifications/notifications.tsx [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    return () => clearTimeout(timer);
+  }, [isAllSeen, news]);
+
+  usePageAnalytic(ScreensEnum.Notifications);
+
+  return (
+    <ScrollView style={styles.contentWrapper}>
+      {isShowPlaceholder ? (
+        <DataPlaceholder text="Notifications not found" />
+      ) : (
```
Comment (unixvb) [inline src/screens/notifications/notifications.tsx:38]: you can use FlatLisit ListEmptyComponent prop instead

File: src/screens/wallet/wallet.tsx [hunk 8]
```diff
@@ -46,6 +48,11 @@ export const Wallet = () => {
           <Divider />
 
           <TouchableIcon name={IconNameEnum.QrScanner} onPress={() => navigate(ScreensEnum.ScanQrCode)} />
+          <Divider size={formatSize(24)} />
+          <TouchableIcon
+            name={!isDotVisible ? IconNameEnum.NotificationsUnread : IconNameEnum.NotificationsRead}
```
Comment (unixvb) [inline src/screens/wallet/wallet.tsx:53]: omit using false positive condiitons

File: src/store/news/news-epics.ts [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+const loadMoreNewsEpic = (action$: Observable<Action>, state$: Observable<RootState>) =>
+  action$.pipe(
+    ofType(loadMoreNewsAction.submit),
+    withLoadedNews(state$),
+    switchMap(([, lastNews]) =>
+      from(
+        getNewsItems({
+          platform: PlatformType.Extension,
+          timeLt: new Date(lastNews[lastNews.length - 1].createdAt).getTime().toString()
+        })
```
Comment (unixvb) [inline src/store/news/news-epics.ts:36]: maybe move `from` operator into `getNewsItems$`?

File: src/store/news/news-reducers.ts [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+    ),
+    loading: false
+  }));
+  builder.addCase(loadMoreNewsAction.success, (state, { payload }) => ({
+    ...state,
+    news: unique([...state.news, ...payload.map(x => ({ ...x, status: StatusType.New }))], 'id')
+  }));
+});
+
+function unique(array: NewsNotificationInterface[], propertyName: keyof NewsNotificationInterface) {
```
Comment (unixvb) [inline src/store/news/news-reducers.ts:38]: could u use `import { uniq } from 'lodash-es';` instead?

File: src/store/news/news-state.ts [hunk 11]
```diff
@@ -0,0 +1,18 @@
+import { EmptyObject } from '@reduxjs/toolkit';
+
+import { NewsNotificationInterface } from '../../interfaces/news.interface';
+import { welcomeNewsNotifications } from './news-data';
+
+export interface NewsState {
+  news: Array<NewsNotificationInterface>;
+  loading: boolean;
```
Comment (unixvb) [inline src/store/news/news-state.ts:8]: use LoadableEntityState instead

File: src/store/news/news-reducers.ts (hunk 10 above)
Comment (letier3110) [reply src/store/news/news-reducers.ts:38]: only `uniqBy`

File: src/screens/notifications/icons/news.icon.tsx [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+    d="M18.75 18H9.75M12.75 6H9.75H12.75ZM12.75 9H9.75H12.75ZM18.75 12H9.75H18.75ZM18.75 15H9.75H18.75Z"
+    stroke="#FF5B00"
+    stroke-linecap="round"
+    stroke-linejoin="round"
+  />
+  <path
+    d="M15.75 9.25H18.75C18.8163 9.25 18.8799 9.22366 18.9268 9.17678L19.2803 9.53033L18.9268 9.17678C18.9737 9.12989 19 9.0663 19 9V6C19 5.9337 18.9737 5.87011 18.9268 5.82322C18.8799 5.77634 18.8163 5.75 18.75 5.75H15.75C15.6837 5.75 15.6201 5.77634 15.5732 5.82322L15.2197 5.46967L15.5732 5.82322C15.5263 5.87011 15.5 5.9337 15.5 6V9C15.5 9.0663 15.5263 9.12989 15.5732 9.17678L15.2197 9.53033L15.5732 9.17678C15.6201 9.22366 15.6837 9.25 15.75 9.25Z"
+    stroke="#FF5B00"
+  />
+</svg>;
```
Comment (unixvb) [inline src/screens/notifications/icons/news.icon.tsx:72]: what is this?

File: src/screens/notifications/icons/update.icon.tsx [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+  />
+  <path d="M12 19H12.01" stroke="#FF5B00" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" />
+  <path
+    fill-rule="evenodd"
+    clip-rule="evenodd"
+    d="M11.9027 7.12381C11.9516 7.10787 12.0038 7.09924 12.0579 7.09924C12.1989 7.09924 12.3263 7.1576 12.4172 7.25147L15.8701 10.7044C16.0654 10.8997 16.0654 11.2163 15.8701 11.4115C15.6749 11.6068 15.3583 11.6068 15.163 11.4115L12.5579 8.80644L12.5579 14.5167C12.5579 14.7928 12.3341 15.0167 12.0579 15.0167C11.7818 15.0167 11.5579 14.7928 11.5579 14.5167L11.5579 8.80626L8.95267 11.4115C8.75741 11.6068 8.44083 11.6068 8.24557 11.4115C8.05031 11.2163 8.05031 10.8997 8.24557 10.7044L11.7043 7.24569C11.7619 7.18805 11.8302 7.14742 11.9027 7.12381Z"
+    fill="#FF5B00"
+  />
+  <path d="M9 3H15C15 3.55228 14.5523 4 14 4H10C9.44772 4 9 3.55228 9 3Z" fill="#FF5B00" />
+</svg>;
```
Comment (unixvb) [inline src/screens/notifications/icons/update.icon.tsx:62]: and this

File: src/screens/notifications/notifications.tsx (hunk 7 above)
Comment (unixvb) [reply src/screens/notifications/notifications.tsx:38]: not resolved

File: src/screens/notifications/icons/update.icon.tsx (hunk 13 above)
Comment (letier3110) [reply src/screens/notifications/icons/update.icon.tsx]: artifacts from state transition of svg

**Review by unixvb — APPROVED** [review]

**Review by LendiHop — APPROVED** [review]

**Review by herkoss — APPROVED** [review]

**Review by oryhivskiy — APPROVED** [review]

**Pull request merged**